src/iterators/drop-iterator.ts

```diff
@@ -0,0 +1,80 @@
+import { BaseIterator, DONE } from './base-iterator.js';
+
+export class DropIterator<T> extends BaseIterator<T> {
+	#count!: number;
```

Comment:
why is `!:` it seems like you assign it definitely in the constructor

Reply:
I do this in 3 places: `drop()`, `take()`, and in `concat()`. I move the assignment above the validation in both `drop()` and `take()`.

I don't want to/cant't do that for `concat(other)` as I would need to make the property that holds the other iterable be `undefined`, which it will never be, then I'd have assert the iterable is valid every where when I know it will be post constructor.